{
  if (isDirty()) {
    OLogManager.instance().warn(this,"Storage " + name + " was not closed properly. Will try to restore from write ahead log.");
    try {
      wereDataRestoredAfterOpen=restoreFromWAL() != null;
    }
 catch (    Exception e) {
      OLogManager.instance().error(this,"Exception during storage data restore",e);
      throw e;
    }
    OLogManager.instance().info(this,"Storage data restore was completed");
  }
}
